tests/vmi_lifecycle_test.go

```diff
@@ -93,6 +93,29 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 		tests.EnsureKVMPresent()
 	})
 
+	Context("when virt-launcher is deleted", func() {
+		It("should label the node with kubevirt.io/schedulable=false", func() {
+			pods, err := virtClient.CoreV1().Pods("").List(metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("%s=%s", v1.AppLabel, "virt-handler"),
+			})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pods.Items).ToNot(BeEmpty())
+
+			pod := pods.Items[0]
+			handlerNamespace := pod.GetNamespace()
+			err = virtClient.CoreV1().Pods(handlerNamespace).Delete(pod.Name, &metav1.DeleteOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			Eventually(func() string {
+				n, err := virtClient.CoreV1().Nodes().Get(pod.Spec.NodeName, metav1.GetOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				return n.Labels[v1.NodeSchedulable]
+			}, 20*time.Second, 1*time.Second).Should(Equal("false"))
+
```

Comment:
I think that we need another test that checks what happens when there is no graceful shut-down of virt-handler.

Reply:
This test exists already. Our node-controller in the virt-controller binary takes care of this. It watches the nodes and sees if virt-handlers heartbeat is gone.

Reply:
:+1: thanks for the info